fs/socket/socket.c

```diff
@@ -285,6 +284,9 @@ int socket(int domain, int type, int protocol)
 errout_with_sockfd:
   nx_close(sockfd);
```

Comment:
doesn't this free the psock and causes double free below?

Reply:
Done